src/EditorFeatures/CSharp/FixInterpolatedVerbatimString/FixInterpolatedVerbatimStringCommandHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Utilities;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.FixInterpolatedVerbatimString
+{
+    [Export(typeof(VSCommanding.ICommandHandler))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [Name(nameof(FixInterpolatedVerbatimStringCommandHandler))]
+    internal sealed class FixInterpolatedVerbatimStringCommandHandler : IChainedCommandHandler<TypeCharCommandArgs>
+    {
+        public string DisplayName => CSharpEditorResources.Fix_interpolated_verbatim_string;
+
+        public void ExecuteCommand(TypeCharCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+        {
+            // We need to check for the token *after* the opening quote is typed, so defer to the editor first
+            nextCommandHandler();
+
+            if (args.TypedChar == '"')
+            {
+                var caret = args.TextView.GetCaretPoint(args.SubjectBuffer);
```

Comment:
Is there any way to support the new multi-caret functionality?

Reply:
/cc @gundermanc

Reply:
@JPWilli is the multi carets expert.

Try this: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.editor.textviewextensions.getmultiselectionbroker?view=visualstudiosdk-2017

The [IMultiSelectionBroker](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.imultiselectionbroker?view=visualstudiosdk-2017) interface is used to get and set caret positions. Much of [IEditorOptions])https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.operations.ieditoroperations3?view=visualstudiosdk-2017) has also been updated to correctly handle multiple carets.

Reply:
Thanks, I will look into this later.

Reply:
I can't find this API. I assume we aren't referencing 15.8 VS assemblies yet? Maybe this can be revisited later then.

Reply:
![image](https://user-images.githubusercontent.com/11444821/45886923-a6837580-bdba-11e8-9a3f-186241b830d5.png)

Reply:
yes, multicaret support was only added in 15.8